### DISCUSSION TITLE
QT Py ESP32-C3 not connecting to wifi

### DISCUSSION THREAD

**Discussion by mirshko** [microsoft/devicescript#584] [Q&A] [answered]
I got a couple of ESP32-C3s from Adafruit and have been trying to get them to connect to wifi, I've got a Unifi AP setup with a standard WPA2 network, nothing fancy. 

My setup is an M2 Pro MacBook, latest VSCode / Extension / Node LTS, etc.. and on both devices nothing works. 

They both seem to run the code fine but any wifi-related stuff seems to fail, just get loads of `lostip` events. 

Additionally they fail to move to the Wifi provisioning step when trying to use the whippersnapper bootloader flashed with esptool, so I'm wondering if they have bad wifi chips or what?

**Comment by pelikhan** [comment]
The Qt Py esp32-c3 is definitely a board we've been testing on.

If you are able to flash other samples using Wifi with Arduino or Micropython, you should be able to rule out hardware malfunction.

**Reply by pelikhan** [reply]
Do you have the devicescript output? The wifi stack has quite a bit of verbose output that might help diagnose the issue.

**Reply by mirshko** [reply]
Sorry what do you mean by output? I only found the console and the debug stuff in the extensions panel.

**Reply by pelikhan** [reply]
Yes sorry, the terminal output typically has a lot of information about the state of the wifi stack. Are you able to connect to WiFi with something else than DeviceScript?

**Reply by mirshko** [reply]
I tried with multiple Arduino samples on my windows pc and nothing, while the boards can scan the wifi around, they can't connect to anything, not even my phones hotspot, so I'm rather confused why they can scan and see networks but not connect. I noticed my Access Points caused multiple SSIDs with the same name but disabling all but one still cause the ESPs to not connect. Dunno if I have a bad batch or what now, but clearly it's not a device script issue, or OS issue.

**Reply by pelikhan** [reply]
Adafruit might more magic debugging tips for you.

**Reply by mirshko** [reply]
Quite likely! Thanks for the help though!

**Reply by mirshko** [reply, marked as answer]
Found this thread on the Adafruit forums, seems the C3 is known to have issues and they’re issuing refunds. 

https://forums.adafruit.com/viewtopic.php?t=202421

And another: https://forums.adafruit.com/viewtopic.php?t=202978

**Reply by mmoskal** [reply]
This is interesting. I designed my own C3 board that had the same issue. I thought I somehow messed up the uFL antenna connector routing but maybe not :)

**Reply by mirshko** [reply]
If it’s not just an Adafruit issue maybe it’s an Espressif issue. ¯\_(ツ)_/¯ seems the S2 / S3 Are Fine, maybe the C6 is better? It also has wifi 6 so that’s neat

**Reply by louisvangeldrop** [reply]
I had the same problem with a Lilygo TTGO board see: https://github.com/Xinyuan-LilyGO/LilyGo-T-OI-PLUS/issues/4
Lilygo shipped me a newer version of the board and that solved the problem. I got a refund for the old board.
